app/sockets/service.py

```diff
@@ -128,7 +165,26 @@ def conversation_init(raw_data, data=None, user=None, error=None):
                 SocketService.add_conversation_identifier(user['email'], request.sid)
             except InvalidOperationException, e:
                 emit('error', e.message)
-        
+
+        @socketio.on('disconnect', namespace="/global")
+        @authenticated_event
+        def global_disconnect(raw_data, data=None, user=None, error=None):
+            if not user:
+                emit('error', "Authentication is required to disconnect this namespace")
+            if error:
+                emit('error', error)
+                
+            SocketService.remove_identifier(user['email'])
+
+        @socketio.on('disconnect', namespace="/conversation")
+        @authenticated_event
+        def conversation_disconnect(raw_data, data=None, user=None, error=None):
+            if not user:
+                emit('error', "Authentication is required to disconnect this namespace")
+            if error:
+                emit('error', error)
+
+            SocketService.leave_conversation(request.sid, request.namespace, user['email'])
```

Comment:
I'm not sure that this will work, but I don't think that we need this anyways. As you said, SocketIO removes users from rooms on disconnect anyways, right?